web/modules/custom/dpl_event/src/PriceFormatter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\dpl_event;
+
+use Brick\Math\BigDecimal;
+use Drupal\Core\StringTranslation\TranslationInterface;
+
+/**
+ * Formats prices according to local rules.
+ *
+ * This is important as we do not rely on fields built for prices. Instead we
+ * have double fields with values as strings.
+ */
+class PriceFormatter {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    protected TranslationInterface $translation,
+  ) {}
+
+  /**
+   * Format a single price.
+   */
+  public function formatPrice(string $price_string): string {
+    $context = ['context' => 'dpl_event'];
+
+    $price = BigDecimal::of($price_string);
+    return match(TRUE) {
+      // Events with 0 cost should show "Free" instead of a numeric price.
+      $price->isEqualTo(0) => $this->translation->translate("Free", [], $context),
```

Comment:
does this also work if someone has written 0.00?

Reply:
Yes it does, because `BigDecimal` handles this complexity for os. The unit test also shows that this is the case.